util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java

```diff
@@ -236,12 +234,17 @@ protected final Helper getHelper() {
 
   @VisibleForTesting
   public final Collection<ChildLbState> getChildLbStates() {
-    return childLbStates.values();
+    return childLbStates;
   }
 
   @VisibleForTesting
   public final ChildLbState getChildLbState(Object key) {
-    return childLbStates.get(key);
+    for (ChildLbState state : childLbStates) {
```

Comment:
In `ClusterManagerLoadBalancer.acceptResolvedAddresses()`, it iterates over all of the previous children calling getChildLbState.  That now becomes an O(N-squared) operation.  Given that it is only when there are name resolver updates it isn't critical, but if there are 1000's of endpoints even just once every 15 seconds is somewhat slowing down.  It might be good to build a map in CMLB.aRA() to keep that activity linear.

Reply:
https://github.com/grpc/grpc-java/pull/11571 removed that call to `getChildLbState()`. Only tests should be calling this method at this point. I see no usages of `getChildLbState()` nor `getChildLbStateEag()` outside of tests.

Reply:
Sorry